catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,83 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            if (ex.getWarnings() != null) {
```

Comment:
You're right, the previous commit did not generate warnings or errors, this is why I added the try/catch and a different schematron ruleset and metacard. To be clearer, there are no errors here but there is a warning generated in the exception which we verify is sanitized by checking it against a fresh sanitized copy of itself.